File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dit ={}
        for i, ch in enumerate(nums):
            rest = target-ch
            if rest in dit:
                return [dit[rest],i]
            else:
                dit[ch]=i
        return None

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0; i<nums.length; i++) {
            int rest = target-nums[i];
            if(map.containsKey(rest)){
                return new int[] {map.get(rest), i};
            }
            map.put(nums[i],i);
        }
        return null;
    }
}